[16.0][MIG] stock_move_backdating

Saran440 opened the pull request:
This PR migrated from version15 and enhance to supported backdate scrap

tafaRU committed:
[ADD] stock_move_backdating

tafaRU committed:
[FIX]stock_move_backdating: stock_view.xml

eLBati committed:
[FIX] authors

tafaRU committed:
[REF] stock_partial_picking_line

tafaRU committed:
[FIX] changed view

eLBati committed:
[imp] use back date for accounting

eLBati committed:
[imp] orm.Model

eLBati committed:
[imp] use move date and related period

eLBati committed:
[add] missing account.py

tafaRU committed:
[FIX] stock_partial_picking_line

tafaRU committed:
[ADD] override to _partial_move_for and to do_partial in stock.partial.picking

eLBati committed:
[fix] wrong create return value

tafaRU committed:
[FIX] remove pdb

tafaRU committed:
[ADD] AUTHORS and copyright

tafaRU committed:
[ADD] mail address to AUTHORS.txt

tafaRU committed:
[FIX] PEP 8 and OpenERP syntax compliance

eLBati committed:
[FIX] using right date: move.date_backdating

tafaRU committed:
[FIX]replaced m by move and avoid extra spaces

eLBati committed:
[FIX] when move.date_backdating is not set

eLBati committed:
[IMP] stock_move_backdating and stock_picking_invoice_link descriptions

gurneyalex committed:
mark unported module as uninstallable

gurneyalex committed:
Update travis configuration

use sudo: false to enable container build
use cache: pip to cache pip packages

archetipo committed:
porting of the module [FIX] tests [FIX] translation

archetipo committed:
update quants date [FIX] test

eLBati committed:
FIX when using module with user != admin. Test added

sbidoul committed:
[MOV] move addons out of __unported__ (they remain not installable)

archetipo committed:
README [FIX] Travis.yaml [FIX] rebased with 8.0

archetipo committed:
author

archetipo committed:
Wizard date comparison [RM] useless inheritance class stock_quant and method _create_account_move_line [FIX] upgrade quant_ids date in a single call

archetipo committed:
stock logistic workflow available addons

archetipo committed:
travis remarks

archetipo committed:
version number

archetipo committed:
Copyright in py module

oca-transbot committed:
OCA Transbot updated translations from Transifex

pedrobaeza committed:
[MIG] Rename manifest files

tafaRU committed:
stock_move_backdating module (#252)

* Apply some cosmetic changes according to OCA guidelines

* Add date_backdating to stock.transfer.details and set date_done of stock.picking as the youngest date among moves

* Update test accordingly to last changes

* Update i18n/it.po accordingly to last changes

* Apply some cosmetic changes: copyright and OCA url

* [IMP] stock_move_backdating: decription

* FIX stock_move_backdating: avoid to use future date anyway, if forced in web interface

tafaRU committed:
stock_move_backdating: fix TestStockMoveBackdating

oca-transbot committed:
OCA Transbot updated translations from Transifex

sbidoul committed:
fix website url in manifest

tafaRU committed:
[MIG] stock_move_backdating: Migration to 10.0

tafaRU committed:
[ADD] stock_move_backdating: add 'Fill Actual Movement Date' button

tafaRU committed:
[FIX] stock_move_backdating: avoid searching in a loop and filter only done moves

tafaRU committed:
[FIX] stock_move_backdating: prevent check_date for empty date

tafaRU committed:
[IMP] stock_move_backdating: add check_date to FillDateBackdating wizard

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 88.2% (15 of 17 strings)

Translation: stock-logistics-workflow-10.0/stock-logistics-workflow-10.0-stock_move_backdating
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-10-0/stock-logistics-workflow-10-0-stock_move_backdating/pt/

SirTakobi committed:
[MIG] stock_move_backdating: Migration to 12.0

SirTakobi committed:
[IMP] stock_move_backdating: Update source quant date

SirTakobi committed:
[IMP] stock_move_backdating: Show date in picking

SirTakobi committed:
[I18N] stock_move_backdating: Update Italian and POT

SirTakobi committed:
[IMP] stock_move_backdating: Allow to backdate inventory

SirTakobi committed:
[IMP] stock_move_backdating: Performance

Co-authored-by: Denis Roussel <denis.roussel@acsone.eu>

Commit:
[UPD] Update stock_move_backdating.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
stock_move_backdating 12.0.1.0.1

mymage committed:
Translated using Weblate (Italian)

Currently translated at 95.4% (21 of 22 strings)

Translation: stock-logistics-workflow-12.0/stock-logistics-workflow-12.0-stock_move_backdating
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-12-0/stock-logistics-workflow-12-0-stock_move_backdating/it/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 95.4% (21 of 22 strings)

Translation: stock-logistics-workflow-12.0/stock-logistics-workflow-12.0-stock_move_backdating
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-12-0/stock-logistics-workflow-12-0-stock_move_backdating/it/

ps-tubtim committed:
[IMP] stock_move_backdating: black, isort, prettier

ps-tubtim committed:
[MIG] stock_move_backdating: Migration to 15.0

OCA-git-bot committed:
[UPD] README.rst

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: stock-logistics-workflow-12.0/stock-logistics-workflow-12.0-stock_move_backdating
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-12-0/stock-logistics-workflow-12-0-stock_move_backdating/it/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: stock-logistics-workflow-12.0/stock-logistics-workflow-12.0-stock_move_backdating
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-12-0/stock-logistics-workflow-12-0-stock_move_backdating/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: stock-logistics-workflow-12.0/stock-logistics-workflow-12.0-stock_move_backdating
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-12-0/stock-logistics-workflow-12-0-stock_move_backdating/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (22 of 22 strings)

Translation: stock-logistics-workflow-12.0/stock-logistics-workflow-12.0-stock_move_backdating
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-workflow-12-0/stock-logistics-workflow-12-0-stock_move_backdating/it/

toita86 committed:
[IMP] stock_move_backdating: black, isort, prettier

toita86 committed:
[MIG] stock_move_backdating: Migration to 14.0

Commit:
[UPD] Update stock_move_backdating.pot

OCA-git-bot committed:
[BOT] post-merge updates

Saran440 committed:
[MIG] stock_move_backdating: Migration to 16.0

Saran440 committed:
[IMP] stock_move_backdating: add scrap back date & update readme

rousseldenis commented:
/ocabot migration stock_move_backdating

rousseldenis commented:
@Saran440 Could you fix pre-commit and enhance your commit messages ? Thanks

alessandro-fiorino commented:
I've made a pull to fix the pre-commit errors https://github.com/ecosoft-odoo/stock-logistics-workflow/pull/2

Saran440 commented:
@alessandro-fiorino Thank you for your contribution.
Now, I fixes it. Could you help review please?

alessandro-fiorino commented:
It fails some test. I think the problem is in the test itself https://github.com/ecosoft-odoo/stock-logistics-workflow/blob/ca8e33d35f67f5716e349230eda9285074873212/stock_move_backdating/tests/common.py#L66
`standard_price` is enabled only if the product has variants.

Saran440 commented:
Updated:
- Change method update date in account move from `_backdating_account_moves` to `_account_entry_move`, Fixed for case backdating previous month it will error doesn't match date and sequence move
- Fixed test script

arielbarreiros96 reviewed (commented):
I've reviewed if from a functional perspective and it looks good, I tested mainly the backdating at the moment of making inventory adjustments. I do think the Readme file could include a Usage section, I literally had to read the code to know what to do.

bosd commented:
@arielbarreiros96 Is this one ok for you now? Can you update your review?

Saran440 commented:
@arielbarreiros96 I updated the README. Can you check it?

bosd commented:
@Saran440 Can you fix this one?
Would like to start working on the migration.

Saran440 commented:
@bosd it error from module `stock_valuation_layer_usage`
![Selection_017](https://github.com/user-attachments/assets/a925839f-c08d-4672-9f19-e120efd5340a)

I never use this module. So, i can't fixed it.

Dranyel-Bosd reviewed (approved):
LGTM :100:

bosd commented:
v18 migration in stock_move_backdating #1921

bosd reviewed (changes requested):
While working on the v18 migration I noticed some unit tests are disabled and needs fixing. (Fixes are applied in V18 migration)

The issue is solely with the unit tests. The module itself is working perfectly. :+1:

arielbarreiros96 commented:
> @arielbarreiros96 I updated the README. Can you check it?

Thanks for the update. Checked README after edits, seems much better now, simple and pretty straight forward👌

supakornkim commented:
V16 LGTM!

Saran440 committed:
[FIX] stock_move_backdating: update date backdating when stock is backdate only

Saran440 commented:
Fixed error test script.

@map-240221 @bosd Could you check please? and cherry-pick commit 4bfea2d95b9ed0402e7c8c3c66b0b3cf6b1911a7 to version17, 18

map-240221 commented:
> [4bfea2d](https://github.com/OCA/stock-logistics-workflow/commit/4bfea2d95b9ed0402e7c8c3c66b0b3cf6b1911a7)

Works fine here, added to the version 17 migration.

bosd commented:
Can we get a merge?

pere-aquarian commented:
Not an expert, guys. But do you need any help to make codecov work?

Saran440 commented:
@pere-aquarian Thanks for your notice. However, I think v16 is outdated for updating the test scripts. This should be improved in the newer version.

bosd commented:
@rousseldenis Can you please merge?

bosd commented:
/ocabot rebase
